webpack.config.js

```diff
@@ -28,42 +28,69 @@ const commonEntryFiles = {
     background: [path.resolve(__dirname, 'src/background/background-init.ts')],
 };
 
-const commonConfig = {
-    entry: commonEntryFiles,
-    module: {
-        rules: [
-            {
-                test: /\.tsx?$/,
-                use: [
-                    {
-                        loader: 'ts-loader',
-                        options: {
-                            transpileOnly: true,
-                            experimentalWatchApi: true,
+const electronEntryFiles = {
+    main: [path.resolve(__dirname, 'src/electron/main/main.ts')],
+    injected: [path.resolve(__dirname, 'src/injected/stylesheet-init.ts'), path.resolve(__dirname, 'src/injected/client-init.ts')],
+};
+
+const createCommonConfig = entry => {
+    const baseConfig = {
+        entry,
+        module: {
+            rules: [
+                {
+                    test: /\.tsx?$/,
+                    use: [
+                        {
+                            loader: 'ts-loader',
+                            options: {
+                                transpileOnly: true,
+                                experimentalWatchApi: true,
+                            },
                         },
-                    },
-                ],
-                exclude: ['/node_modules/'],
-            },
-        ],
-    },
-    resolve: {
-        modules: [path.resolve(__dirname, 'node_modules')],
-        extensions: ['.tsx', '.ts', '.js'],
+                    ],
+                    exclude: ['/node_modules/'],
+                },
+            ],
+        },
+        resolve: {
+            modules: [path.resolve(__dirname, 'node_modules')],
+            extensions: ['.tsx', '.ts', '.js'],
+        },
+        plugins: commonPlugins,
+        node: {
+            setImmediate: false,
+        },
+        performance: {
+            // We allow higher-than-normal sizes because our users only have to do local fetches of our bundles
+            maxEntrypointSize: 10 * 1024 * 1024,
+            maxAssetSize: 10 * 1024 * 1024,
+        },
+    };
+
+    return baseConfig;
+};
+
+const electronConfig = {
+    ...createCommonConfig(electronEntryFiles, true),
+    name: 'electron',
+    mode: 'development',
+    output: {
+        path: path.join(__dirname, 'extension/electronBundle'),
+        filename: '[name].bundle.js',
     },
-    plugins: commonPlugins,
     node: {
-        setImmediate: false,
+        __dirname: false,
+        __filename: false,
     },
-    performance: {
-        // We allow higher-than-normal sizes because our users only have to do local fetches of our bundles
-        maxEntrypointSize: 10 * 1024 * 1024,
-        maxAssetSize: 10 * 1024 * 1024,
+    optimization: {
+        splitChunks: false,
     },
+    target: 'electron-main',
 };
 
 const devConfig = {
-    ...commonConfig,
```

Comment:
Would it be more consistent with how the individual configs customize other config-specific properties to do

```ts
const individualConfig = {
    ...commonConfig,
    entry: individualEntryFiles,
    someConfigSpecificProperty: 'individual-setting',
    // ...
}
```

instead of 

```ts
const individualConfig = {
    ...createCommonConfig(individualEntryFiles),
    someConfigSpecificProperty: 'individual-setting',
    // ...
}
```